### ISSUE TITLE
Bug in zooming in non-uniform files

### ISSUE THREAD

**Issue by pqbvojgdeh** [sumatrapdfreader/sumatrapdf#1422]
SumatraPDF 3.2.12052 x64-bit, Windows 10 1809 (17763.973). I checked with the portable version of Sumatra with the default settings on file https://yadi.sk/i/NmCy-Ca-H_MRJw.
My actions:
1. Open the file.
2. Scale (zoom in) the file with the right mouse button pressed and the wheel turned up.
3. On the fifth step (turning the mouse wheel) document scale will be reduced. Then the zoom continues in the normal mode.

OR

1. After opening the file press "+" on the keyboard.
2. Document scale will be reduced.
3. Then the zoom continues in the normal mode.

**Comment by GitHubRulesOK** [comment]
@kjk I can confirm this unusual behaviour with this file is still present in current build 12063

In general it seems to be confined to this type of file with mixed portrait and landscape pages?
Portrait OR Landscape ONLY files tested do not seem to trigger the same behaviour!
The behaviour repeatedly appears when the file is first opened with "fit width" initially active on the first portrait page then a zoom in (most easily by pressing +) appears to produce a "66.67%" zoom before continuing to zoom in normally.
Most likely the first action step is to recalculate the page width to accommodate the landscape width then zoom in one step from that wider view.

In summary an understandable if unwanted feature of "fitting" the first page width

**Comment by kjk** [comment]
The issue is caused by switching from `fit width` mode which, for this file, has a different zoom level for each page, to a mode where zoom is the same for all pages.

Maybe could be improved although there's always be a weird effect when switching between variable and fixed zoom levels.

**Comment by pqbvojgdeh** [comment]
Do you have any ideas to fix this bug? It is extremely inconvenient to use scaling in such documents.
